test/hack/bin/verify-rules.sh

```diff
@@ -0,0 +1,200 @@
+#!/bin/bash
+set -euo pipefail
+#
+# This test ensures that all prometheus rules are valid
+#
+
+set -eu
+
+array_contains() {
+    local search="$1" && shift
+
+    local element
+    for element; do
+        if [[ "$element" == "$search" ]]; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+main() {
+
+    START_TIME="$(date +%s)"
+    echo "$(date '+%H:%M:%S') promtool: start"
+
+    local GIT_WORKDIR
+    GIT_WORKDIR=$(git rev-parse --show-toplevel)
+
+    local PROMTOOL=test/hack/bin/promtool
+    local HELM=test/hack/bin/helm
+    local ARCHITECT=test/hack/bin/architect
+    local YQ=test/hack/bin/yq
+
+    # prepare the helm chart
+    echo "### Helm chart preparation"
+    "$GIT_WORKDIR/$ARCHITECT" helm template --dir "$GIT_WORKDIR/helm/prometheus-rules" --dry-run
+
+    expected_failure_relative_file_global="test/conf/promtool_ignore"
+    expected_failure_file_global="$GIT_WORKDIR/$expected_failure_relative_file_global"
+
+    # Retrieve all files we're going to check
+    local -a all_files
+    mapfile -t all_files < <(
+        cd "$GIT_WORKDIR" || return 1
+        # filter alerting-rules files, and remove prefix `helm/prometheus-rules/`
+        git ls-files |
+            sed -En 's_^helm/prometheus-rules/(templates/alerting-rules/.*\.ya?ml)$_\1_p' || echo error
```

Comment:
That's a bit hidden, and why are we ignoring alerting rules here ?

Reply:
`sed -En 's_^helm/prometheus-rules/(templates/alerting-rules/.*\.ya?ml)$_\1_p'` search for all alerting rules (so we exclude all other files expect the alerting rules) and removes the `helm/prometheus-rules` prefix. This is used to run `helm template` for only this file with different providers.

Reply:
i aggree with you, that we should also check the `recording-rules` directory - but i would recommend to implement this after this PR got merged. The testing stuff was primary implemented for alerting-rules and i'm not 100% sure yet how to unit test recording rules.

Reply:
recording rules are the same things than alerting rules, same format, same resources, just different purpose. But fine let's only test alerting rules for now. This is already a huge improvement.

Reply:
I documented in the readme that we only test alerting-rules.